src/ast.mli

```diff
@@ -1,7 +1,83 @@
-type operator = Add | Sub | Mul | Div
+type bin_op =
+    | Add    | Minus | Multiply | Divide
+    | Modulo | Equal | Neq      | Lt
+    | Leq    | Gt    | Geq      | And
+    | Or
+
+type u_op = Not | Neg
+
+type p_type = Int_t | Bool_t | Double_t | Char_t | Unit_t
+
+type cont_type = String_t | List_t | Set_t | Map_t | Tuple_t
+
+type opt_type = Maybe_t of types | Some_t of types | None_t
+
+type message_type = Message_t
+
+type field = Field of types * string
+
+type func = {
+    func_name : string;
+    function_return_t : types;
+    function_formals : formal list;
+    function_body : stmt list;
+}
+
+type actor = {
+    actor_fields : field list;
+    actor_name : string;
+    acator_receive: receive;
+}
+
+type actor_type = Actor_t of string
+
+type pool_type = Pool_t of actor_type list
+
+type types =
+    | Primitive of p_type
+    | Container of cont_type
+    | Optional of opt_type
+    | Actor_t of actor_op
```

Comment:
Is `Actor_t` a special convention? It does not conform to the other names here, and I'd rather we be uniform, i.e. change this to `Actor`.